rs/execution_environment/src/scheduler/round_schedule.rs

```diff
@@ -366,7 +363,8 @@ impl RoundSchedule {
 
             let has_aborted_or_paused_execution =
                 canister.has_aborted_execution() || canister.has_paused_execution();
-            if !has_aborted_or_paused_execution {
+            let good_standing = canister.scheduler_state.accumulated_priority >= 0.into();
+            if !has_aborted_or_paused_execution || !good_standing {
                 RoundSchedule::apply_priority_credit(canister);
             }
```

Comment:
Is this actually needed? It feels to me like the only required change is the one below (bundle negative AP DTS executions with non-DTS executions instead of with the other DTS executions).

Reply:
To treat canisters with long executions just like canisters with new executions we need also to charge them every round. But this change does not change much...